### PR TITLE
Enable all end2end tests on CPU/GPU with -d option

### PR DESCRIPTION
**Pull request by mingmingtasd**
I add "-d" option to select devices when running end2end tests. All tests can run with no regression both on CPU and GPU for IE/DML. PTAL, thanks! @fujunwei
@BruceDai Does this solution meet your requirements of testing? Any suggestions? Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,9 +75,20 @@ Then use `ninja -C out/Release` or `ninja -C out/Debug` to build WebNN-native.
...
-Run end2end tests, for example `./out/Release/webnn_end2end_tests`.
+Run end2end tests on a Default device:
+```sh
+> ./out/Release/webnn_end2end_tests
+```
+You can also specify a device to run end2end tests using "-d" option, for example:
+```sh
+> ./out/Release/webnn_end2end_tests -d GPU
+```
+Currently "CPU", "GPU" and "Default" are supported, more devices are to be supported in the future.
```
Comment (BruceDai) [inline README.md:91]: @mingmingtasd What does “Default” mean? Thanks.

Comment (mingmingtasd) [reply README.md:91]: If set -d Default or without -d option, tests will run on a default device.
Default represents different devices on different backends, for example:
GPU is the default device on DML, while CPU is the default device on IE. @BruceDai

Comment (BruceDai) [reply README.md:91]: @mingmingtasd  It's clear for me, thanks much for your explanation.

File: src/tests/End2EndTestsMain.cpp [hunk 2]
```diff
@@ -16,7 +16,28 @@
...
+    std::string device = "Default";
+    for (int i = 1; i < argc; ++i) {
+        if (strcmp("-d", argv[i]) == 0 && i + 1 < argc) {
+            device = argv[i + 1];
+        }
+    }
+    if (device != "CPU" && device != "GPU" && device != "Default") {
+        dawn::ErrorLog()
+            << "Invalid device options, only support \"CPU\", \"GPU\" and \"Default\".";
+        return -1;
```
Comment (fujunwei) [inline src/tests/End2EndTestsMain.cpp:28]: remove L.25 - L.29 and add the exception to L.39.

File: src/tests/End2EndTestsMain.cpp [hunk 3]
```diff
@@ -16,7 +16,28 @@
 #include "tests/WebnnTest.h"
 
 int main(int argc, char** argv) {
-    InitWebnnEnd2EndTestEnvironment();
+    std::string device = "Default";
+    for (int i = 1; i < argc; ++i) {
```
Comment (fujunwei) [inline src/tests/End2EndTestsMain.cpp:20]: Do you mean we need to run the end2end multi-times on different device?

Comment (mingmingtasd) [reply src/tests/End2EndTestsMain.cpp:20]: As discussed with @BruceDai , running on one device per time can help to collect and parse tests results, am I right? @BruceDai

Comment (BruceDai) [reply src/tests/End2EndTestsMain.cpp:20]: @mingmingtasd Yes, if we ran test per device(CPU/GPU/...), we could clearly get test result with specified device. If you supported running test with multi supported devices one time, please consider outputting distinguished test result by device which would help test tool to parse output result.

File: src/tests/End2EndTestsMain.cpp (hunk 2 above)
Comment (mingmingtasd) [reply src/tests/End2EndTestsMain.cpp]: Done, PTAL, thanks! @fujunwei

File: src/tests/End2EndTestsMain.cpp [hunk 4]
```diff
@@ -16,7 +16,27 @@
...
+        if (strcmp("-d", argv[i]) == 0 && i + 1 < argc) {
+            device = argv[i + 1];
+        }
+    }
+
+    ml::ContextOptions options;
+    if (device == "CPU") {
```
Comment (fujunwei) [inline src/tests/End2EndTestsMain.cpp:27]: Should we ignore case sensitivity?

**Review by fujunwei — APPROVED** [review]
LGTM with one nit.

File: src/tests/End2EndTestsMain.cpp (hunk 4 above)
Comment (mingmingtasd) [reply src/tests/End2EndTestsMain.cpp:27]: > Should we ignore case sensitivity?

Thanks! I suggest keeping case sensitive to align with OpenVINO native benchmark. @fujunwei

**Comment by mingmingtasd** [comment]
> @mingmingtasd , please help solve the conflicts. Thanks!

Yes, I have solved them, thanks! @huningxin

File: src/tests/End2EndTestsMain.cpp (hunk 4 above)
Comment (huningxin) [reply src/tests/End2EndTestsMain.cpp:27]: I would suggest to to ignore the case and use the lowercase by default, such as "cpu", "gpu" and "default". That aligns to webnn spec.

Comment (mingmingtasd) [reply src/tests/End2EndTestsMain.cpp]: I have updated to ignore the case , PTAL again, thanks! @huningxin @fujunwei

**Review by huningxin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**